Hierarchical Clustering/Implementation.py

```diff
@@ -0,0 +1,62 @@
+# Mall Customer Segmentation: <font color='blue'>K-Means Clustering</font>
+* Importing the libraries
+* Importing the dataset
+* Dataset information (Pandas Profiling)
+* Elbow method to find the optimal number of clusters
+* Training the K-Means model on the dataset
+* Visualising the clusters
+
+## Importing the libraries
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+
+"""## Importing the dataset"""
+
+dataset = pd.read_csv('../input/customer-segmentation-tutorial-in-python/Mall_Customers.csv')
+X = dataset.iloc[:, [3, 4]].values
+
+"""## Dataset information (Pandas Profiling)"""
+
+# Commented out IPython magic to ensure Python compatibility.
+import pandas_profiling as pp
+import warnings
+warnings.filterwarnings('ignore')
+# %matplotlib inline
+
+pp.ProfileReport(dataset, title = 'Pandas Profiling report of "dataset"', html = {'style':{'full_width': True}})
+
+"""## Elbow method to find the optimal number of clusters"""
+
+from sklearn.cluster import KMeans
+wcss = []
+for i in range(1, 11):
+    kmeans = KMeans(n_clusters = i, init = 'k-means++', random_state = 42)
+    kmeans.fit(X)
+    wcss.append(kmeans.inertia_)
+plt.plot(range(1, 11), wcss, c = 'blue')
+plt.title('The Elbow Method')
+plt.xlabel('Number of clusters')
+plt.ylabel('WCSS')
+plt.show()
+
+"""## Training the K-Means model on the dataset"""
+
+kmeans = KMeans(n_clusters = 5, init = 'k-means++', random_state = 42)
+y_kmeans = kmeans.fit_predict(X)
+
+"""## Visualising the clusters"""
+
+plt.scatter(X[y_kmeans == 0, 0], X[y_kmeans == 0, 1], s = 50, c = 'red', label = 'Cluster 1')
```

Comment:
We need a clean simple class-type implementation so that it can be used later,so these plotting and fitting etc needs to be commented out